src/fs/recursive_size.rs

```diff
@@ -0,0 +1,78 @@
+/// Used to represent a the size of a recursive directory traversal.  `None`
+/// should be used when the file does not represent a directory or the recursive
+/// size should not be calculated.
+#[derive(Copy, Clone, Debug)]
+pub enum RecursiveSize {
+    /// Size should not be computed
+    None,
+    /// Size should be computed but has not been computed yet
+    Unknown,
+    /// Size has been computed.  First field is size in bytes and second field
+    /// is size in blocks
+    #[cfg_attr(target_family = "windows", allow(dead_code))]
+    Some(u64, u64),
+}
+
+impl RecursiveSize {
+    /// Returns `true` if `None`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use eza::fs::recursive_size::RecursiveSize;
+    ///
+    /// let x = RecursiveSize::None;
+    /// assert_eq!(x.is_none(), true);
+    ///
+    /// let x = RecursiveSize::Unknown;
+    /// assert_eq!(x.is_none(), false);
+    ///
+    /// let x = RecursiveSize::Some(0, 0);
+    /// assert_eq!(x.is_none(), false);
```

Comment:
We need a ``` here right?
```suggestion
    /// assert_eq!(x.is_none(), false);
    /// ```
```